apps/main/src/llamalend/mutations/repay.mutation.ts

```diff
@@ -22,6 +25,41 @@ export type RepayOptions = {
   userAddress: Address | undefined
 }
 
+const approveRepay = async (market: LlamaMarketTemplate, { userCollateral, userBorrowed, isFull }: RepayMutation) => {
+  if (isFull) {
+    return (await market.fullRepayApprove()) as Hex[]
+  }
+  if (market instanceof LendMarketTemplate) {
+    return (await market.leverage.repayApprove(userCollateral, userBorrowed)) as Hex[]
+  }
+  if (market.leverageV2.hasLeverage()) {
+    return (await market.leverageV2.repayApprove(userCollateral, userBorrowed)) as Hex[]
+  }
+  console.assert(!+userCollateral, `userCollateral should be 0 for non-leverage repay`)
```

Comment:
Hmm, maybe we should add an `ASSERT` logging status to `logging.ts`?

Reply:
logging.ts is a mistake imo we lose the source of the messages completely
fyi this throws an error on failure